### PR TITLE
Chart editor

### PR DESCRIPTION
**Pull request by felixmichel**

### CODE REVIEW

**Review by tpreusse — COMMENTED** [review]
On a general note: I think we could diverge a bit more from JSON schema at this point and e.g. include `defaultProps` on every schema instead of a `default` key on individual fields and directly name the custom react component as type for advanced cases. Similarly we could drop unused titles and type definitions. 

An example schema diff:
```diff
  return {
-   title: 'TimeBarConfig',
-   type: 'object',
+   defaultProps: default
    properties: {
      basic: {
        xAxis: {
          title: 'Horizontale Achse',
          properties: {
            x: {
              title: 'Spalte auswählen',
              type: 'string',
              enum: fields,
-             default: defaults.x
            },
            timeFormat: {
-             title: 'Datumsformat (Chart)',
              type: 'AxisFormatDropdown',
-             enum: timeFormats,
-             default: defaults.timeFormat,
-             format: 'dynamicDropdown',
              parent: 'xAxis',
-             timeParseDefault: defaults.timeParse
            },
```

This would allow some simplifications all around and make the default props as easily accessible as value. However this could also be postponed to a clean up PR and is not blocking.

File: src/components/Chart/Bars.js [hunk 1]
```diff
@@ -755,20 +756,303 @@ export const propTypes = {
...
-  barStyle: 'lollipop'
-}
+Lollipop.defaultProps = defaultProps.Lollipop
 
 // Lollipop has additional default props
 Lollipop.wrap = 'Bar'
 
 export default BarChart
+
+export const barEditorSchema = ({
```
[the remark this answers is not in the document]
Comment (felixmichel) [reply src/components/Chart/Bars.js:770]: good idea, i will do this

File: src/components/Chart/Editor/Editor.utils.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
+import { useState } from 'react'
+
+const parseCommaValue = (newValue = '', context) => {
+  return newValue
+    .split(',')
+    .map(d => d.trim())
+    .filter(Boolean)
+    .map(d => (context === 'number' ? (d === 0 ? 0 : +d) : d))
```
[the remark this answers is not in the document]
Comment (felixmichel) [reply src/components/Chart/Editor/Editor.utils.js:8]: I have to check this again, but parsing zero gave me null

[the remark this answers is not in the document]
Comment (felixmichel) [reply src/components/Chart/Editor/Editor.utils.js:8]: thats why i do this so complicated

File: src/components/Chart/Editor/index.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+  }
+
+  const createOnDropdownChange = key => item => {
+    return onChange({ ...value, [key]: item.value || undefined })
+  }
+
+  const timeFormatParser = timeParse(
+    value.timeParse || value.timeFormat || '%Y'
+  )
```
[the remark this answers is not in the document]
Comment (felixmichel) [reply src/components/Chart/Editor/index.js]: I moved it to the TickField, but I don't know yet how I will use the default prop.

**Review by tpreusse — APPROVED** [review]

**Pull request merged**